persist-cli/src/main/java/io/ballerina/persist/cmd/Migrate.java

```diff
@@ -0,0 +1,778 @@
+
+        } catch (BalException e) {
+            errStream.println("Error getting entities: " + e.getMessage());
+        }
+
+        if (!queries.isEmpty()) {
+            try {
+                // Copy the source file to the destination folder
+                Files.copy(currentModelPath, newMigrationPath.resolve(currentModelPath.getFileName()));
+            } catch (IOException e) {
+                errStream.println("Error: Copying file failed: " + e.getMessage());
+            }
+        } else {
+            // Delete the newMigrateDirectory
+            boolean deleteNewMigrateFolder = newMigrateDirectory.delete();
+            if (!deleteNewMigrateFolder) {
+                errStream.println("Error: Failed to delete timestamp folder.");
+            }
+
+            // Delete the migrations directory if it is empty
+            if (migrationsDir.exists() && migrationsDir.isDirectory()) {
+                File[] files = migrationsDir.listFiles();
+                if (files != null && files.length == 0) {
+                    try {
+                        Files.delete(migrationsDir.toPath());
+                    } catch (IOException e) {
+                        errStream.println("Error: Failed to delete migration folder: " + e.getMessage());
+                    }
+                }
+            }
+        }
+    }
+
+    // Returns the path of the bal file in the persist directory
+    private static String getBalFilePath(String projectDirName) {
+        String persistDirName = "persist";
+        String balFileExtension = ".bal";
+
+        // Create a file object representing the project directory
+        File projectDir = new File(projectDirName);
+
+        // Check if the project directory exists
+        if (!projectDir.exists() || !projectDir.isDirectory()) {
+            errStream.println("Error: Project directory not found: " + projectDirName);
+            return null;
+        }
+
+        // Create a file object representing the 'persist' directory
+        File persistDir = new File(projectDir, persistDirName);
+
+        // Check if the persist directory exists
+        if (!persistDir.exists() || !persistDir.isDirectory()) {
+            errStream.println("Error: Persist directory not found in project directory: " + projectDirName);
+            return null;
+        }
+
+        // Check if a .bal file exists in the persist directory
+        File[] balFiles = persistDir
+                .listFiles((dir, name) -> name.toLowerCase(Locale.ENGLISH).endsWith(balFileExtension));
+        if (balFiles == null || balFiles.length == 0) {
+            errStream.println("Error: Bal file not found in persist directory: " + projectDirName);
+            return null;
+        }
+
+        // Get the first .bal file found in the persist directory
+        File balFile = balFiles[0];
+        Path balFilePath = balFile.toPath();
+
+        // Return the path to the .bal file
+        return balFilePath.toString();
+    }
```

Comment:
The same logic is there in the `Generate.java`. https://github.com/ballerina-platform/persist-tools/blob/main/persist-cli/src/main/java/io/ballerina/persist/cmd/Generate.java#L95-L120

Shall we move the logic in the `Generate.java` to the `BalProjectUtils` class and call it in both places?
@kalaiyarasiganeshalingam WDYT?

Reply:
Yes, we can. Also, it helps to identify the verification duplicates.

Reply:
@kalaiyarasiganeshalingam @daneshk I've committed this change :+1: